src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -632,6 +632,15 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         // Firing it manually makes sure it does.
         _BackgroundBrush = RootGrid().Background();
         _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"BackgroundBrush" });
+
+        _isBackgroundLight = _isColorLight(bg);
+    }
+
+    bool TermControl::_isColorLight(til::color bg)
+    {
+        // GH#13450 function to check if the current background color is light
+        // enough to need a dark version of the visual bell indicator
+        return 127 < (bg.r + bg.g + bg.b) / 3;

```

Comment:
A better, but still cheap (to code) approach for calculating the lightness/luma of `bg` would be to use the Rec. 601 coefficients directly without converting to linear RGB first. This:
```cpp
bool TermControl::_isColorLight(til::color bg) noexcept
{
    // GH#13450 function to check if the current background color is light
    // enough to need a dark version of the visual bell indicator
    // This is a poor man's Rec. 601 luma.
    const auto l = 30 * bg.r + 59 * bg.g + 11 * bg.b;
    return l > 12750;
}
```

Reply:
I'm glad you brought this up because I saw something similar in `ColorHelper` and I wasn't sure if I should use it instead.

Reply:
Oh wow, I didn't realize we already had that. It uses the Rec. 709 luma constants as recommended by WCAG 2, but that doesn't necessarily mean they're "better" (that it uses `pow` to bring it into linear RGB is however quite helpful).